dbms/src/DataStreams/FilterBlockInputStream.cpp

```diff
@@ -67,13 +67,15 @@ Block FilterBlockInputStream::readImpl()
     /// Until non-empty block after filtering or end of stream.
     while (true)
     {
-        FilterPtr child_filter = nullptr;
-        res = children.back()->read(child_filter, true);
+        // The child of FilterBlockInputStream is UnorderInputStream typically
+        // which does not support read(FilterPtr & res_filter, bool return_filter),
+        // so we call read() here.
+        res = children.back()->read();
```

Comment:
It looks a bit hacky here, will it have any effect if left as it is?

Reply:
Since `UnorderInputStream`  does not support `read(filter, true)`(and will not support), no need to call  `read(filter, true)`, and we do not have to handle `child_filter` now.

Reply:
ok